### PR TITLE
Create sign-up / sign-in modal

### LINKED ISSUE #238: Create menu component and profile button
### Description
Create sign-up and sig-in modals with authentication provider buttons

### Requirements

- component must be responsible
- dark mode ready
- should include transition visuals
- include active, focus shadows and effects
- `Login`or `SignUp` actions in menu  should open appropriate modal when pressed

### Deliverables

- `WelcomeBackModal` - shown when there's user information is available
- AuthModal` - shown when user info isn't available and user pressed signup/login

### References

image

### PR DESCRIPTION
**Pull request by Ilxom-22**
Closes #238

### CODE REVIEW

File: src/Training.AirBnb.Clone.Frontend/src/App.vue [hunk 1]
```diff
@@ -1,10 +1,32 @@
 <template>
 
   <!--    <main-logo/>-->
-  <home-view/>
+  <home-view @sign-in-up-request="toggleModal"/>
+
+  <sign-in-up-modal :modal-active="modalActive" @close-modal="toggleModal"/>
```
Comment (sulton-max) [inline src/Training.AirBnb.Clone.Frontend/src/App.vue:6]: As we are teleporting `SignInUpModal` to body, why not add it to `MainHeader` component ?

File: src/Training.AirBnb.Clone.Frontend/src/Modules/locations/components/SignInUpModal.vue [hunk 2]
```diff
@@ -0,0 +1,87 @@
+<template>
+
+<Teleport to="body">
+    <div v-show="modalActive" class="fixed inset-0 z-10 overflow-auto bg-black bg-opacity-50 no-scrollbar" @click="closeModal">
+    
+        <div class="relative h-auto mx-auto my-8 border-2 sm:w-full md:w-[570px] lg:w-[570px] xl:w-[570px] theme-bg-secondary theme-border rounded-xl" @click.stop>
```
Comment (sulton-max) [inline src/Training.AirBnb.Clone.Frontend/src/Modules/locations/components/SignInUpModal.vue:6]: Isn't adding md:w-[570px] enough here ?

File: src/Training.AirBnb.Clone.Frontend/src/Modules/locations/components/EmailPhoneInput.vue [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    else 
+        emit('inputSubmit', phoneNumber.value);
+}
+
+watch(() => props.byEmail, (oldValue, newValue) => {
+    if (newValue != oldValue) {
+        email.value = '';
+        phoneNumber.value = '';
+    }
+})
```
Comment (sulton-max) [inline src/Training.AirBnb.Clone.Frontend/src/Modules/locations/components/EmailPhoneInput.vue:84]: I think comparing old and new values is unnecessary here, just pass expression body

**Review by sulton-max — CHANGES_REQUESTED** [review]

File: src/Training.AirBnb.Clone.Frontend/src/common/components/MainHeader.vue [hunk 4]
```diff
@@ -22,9 +25,23 @@
...
+const toggleModal = () => {
+  modalActive.value = !modalActive.value;
+}
+
+watchEffect(() => {
+  if (modalActive.value) {
+    document.body.classList.add('overflow-hidden');
+  } else {
+    document.body.classList.remove('overflow-hidden');
+  }
```
Comment (sulton-max) [inline src/Training.AirBnb.Clone.Frontend/src/common/components/MainHeader.vue:44]: Can we move this to `DocumentService` too ?

File: src/Training.AirBnb.Clone.Frontend/src/Modules/locations/components/EmailPhoneInput.vue [hunk 5]
```diff
@@ -0,0 +1,84 @@
+<template>
+
+<div class="mt-5">
+    <CountryCodePicker v-if="!byEmail" @country-code-changed="onCountryChanged"/>
+
+    <div class="h-[56px] w-auto flex flex-col justify-center px-3 py-1"
+        :class="[isFocused ? 'border-2 rounded-md border-black dark:border-bgColorPrimary' : byEmail ? 'border rounded-md border-[#b0b0b0]' : 'border-x border-b rounded-b-md border-[#b0b0b0]']"
+        tabindex="0">
+        
+        <h3 class="text-[#797979] text-[12px]">{{ byEmail ? 'Email' : 'Phone number' }}</h3>
```
Comment (sulton-max) [inline src/Training.AirBnb.Clone.Frontend/src/Modules/locations/components/EmailPhoneInput.vue:10]: Add colors to tailwind configuration

**Review by sulton-max — CHANGES_REQUESTED** [review]

**Review by rashidovasadbek — APPROVED** [review]

**Review by sulton-max — APPROVED** [review]
Looks good to me

**Review by rashidovasadbek — APPROVED** [review]

**Pull request merged**